src/libstd/sys/windows/net.rs

```diff
@@ -131,9 +132,9 @@ impl Socket {
     pub fn read(&self, buf: &mut [u8]) -> io::Result<usize> {
         // On unix when a socket is shut down all further reads return 0, so we
         // do the same on windows to map a shut down socket to returning EOF.
+        let len = cmp::min(buf.len(), i32::max_value()) as i32;
```

Comment:
oh oops good catch! I also figured it's better to use `i32` here as it's what's actually required for the system call (eg this is windows specific).

I'm not a huge fan of the usage of `wrlen_t` in general unfortunately...